File: scripts/4b_check_restore_status.js
const weaviate = require("weaviate-client");

const client = weaviate.client({
  scheme: 'http',
  host: 'localhost:8090',
});

client.backup.restoreStatusGetter()
  .withBackend("filesystem")
  .withBackupId("my-very-first-backup")
  .do()
  .then(console.log)
  .catch(console.error)

File: scripts/4b_check_restore_status.go
package main

import (
	"context"
	"encoding/json"
	"fmt"

	"github.com/semi-technologies/weaviate-go-client/v4/weaviate"
	"github.com/semi-technologies/weaviate-go-client/v4/weaviate/backup"
)

func main() {
	hostW2 := "localhost:8090"

	ctx := context.Background()
	client := weaviate.New(weaviate.Config{Scheme: "http", Host: hostW2})

	fmt.Println("\nChecking restore status");
	restoreStatusResponse, err := client.Backup().RestoreStatusGetter().
		WithBackupID("my-very-first-backup").
		WithBackend(backup.BACKEND_FILESYSTEM).
		Do(ctx)
	if err != nil {
		fmt.Println(err)
		return
	}
	asJson, _ := json.MarshalIndent(restoreStatusResponse, "", "  ")
	fmt.Printf("%s\n", asJson)
}
